Sources/Sentry/Profiling/SentryContinuousProfiler.mm

```diff
@@ -0,0 +1,76 @@
+#import "SentryContinuousProfiler.h"
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+
+#    import "SentryLog.h"
+#    import "SentryProfiler+Private.h"
+#    include <mutex>
+
+#    pragma mark - Private
+
+namespace {
+/** @warning: Must be used from a synchronized context. */
+std::mutex _unsafe_gContinuousProfilerLock;
+
+/** @warning: Must be used from a synchronized context. */
+SentryProfiler *_Nullable _unsafe_gContinuousCurrentProfiler;
+
+/** @warning: Must be called from a synchronized context. */
+BOOL
+_unsafe_isRunning(void)
+{
+    return _unsafe_gContinuousCurrentProfiler != nil &&
+        [_unsafe_gContinuousCurrentProfiler isRunning];
+}
+} // namespace
+
+@implementation SentryContinuousProfiler
```

Comment:
💯 that you duplicated the implementation of `SentryContinuousProfiler` and `SentryLegacyProfiler`. Although we could add some abstractions and refactor them to have some base implementation, I prefer duplicating them because there are small differences between them, and the code is way easier to read.